simplified-books-borrowing/src/main/java/org/nypl/simplified/books/borrowing/internal/BorrowLoanCreate.kt

```diff
@@ -84,20 +87,36 @@ class BorrowLoanCreate private constructor() : BorrowSubtaskType {
       context.taskRecorder.addAttribute("Loan URI", currentURI.toString())
       context.checkCancelled()
 
+      val credentials = context.account.loginState.credentials
+
+      val auth =
+        AccountAuthenticatedHTTP.createAuthorizationIfPresent(credentials)
+
       val request =
         context.httpClient.newRequest(currentURI)
           .setMethod(Put(ByteArray(0), applicationOctetStream))
-          .setAuthorization(authorizationOf(context.account))
+          .setAuthorization(auth)
+          .addCredentialsToProperties(credentials)
           .build()
 
       return request.execute().use { response ->
         when (val status = response.status) {
-          is LSHTTPResponseStatus.Responded.OK ->
+          is LSHTTPResponseStatus.Responded.OK -> {
+            context.account.updateCredentialsIfAvailable { currentCredentials ->
+              if (currentCredentials is AccountAuthenticationCredentials.BasicToken) {
+                currentCredentials.updateAccessToken(status.getAccessToken())
+              } else {
+                currentCredentials
+              }
+            }
```

Comment:
I think I've read this same snippet of code four times now. It's not great to be having to do this everywhere the app makes a request.